docs/chapters/architecture/DLT/diagnostics-log-and-tracing.rst

```diff
@@ -0,0 +1,177 @@
+
+* `GENIVI dlt-daemon`_
+* `GENIVI dlt-viewer`_
+
+Since GENIVI dlt-daemon supports only C applications, another third party software is required to support C++. Currently, PELUX uses `ivi-logging`_. However, wrapper discussions are still ongoing. We currently have three options which are compared under Design Decisions.
+
+Building blocks view
+--------------------
+
+Level 1
+^^^^^^^
+.. image:: graphs/block-view-lvl1.png
+
+Level 2 
+^^^^^^^
+.. image:: graphs/block-view-lvl2.png
+
+Runtime view
+------------
+Depending on what the developers decide to log and trace, the logs can be viewed via dlt-viewer.
+
+Optionally, Blue PiraT 2 can be configured to work with dlt-daemon, through which multiple gigabytes of logs can be stored.
+
+
+Design decisions
+----------------
+
+C++ wrapper
+^^^^^^^^^^^
+
+`ivi-logging`_
+""""""""""""""
+This component was initially developed by Pelagicore but is not actively maintained at this time.
+
+Pros:
+
+* It exposes all functions of libdlt.so library.
+* C++11
+
+Cons:
+
+* It is a header-based library. Each translation unit (\*.o) will have around 20Kb of code segment overhead. Each binary file will be bloated and heavily flooded with inline code.
+* Increases compilation time of every translation unit where it is used.
+* Nobody is familiar with it
+
+`log4cplus`_
+""""""""""""
+A popular logging library that inherits log4java principles and log formats
+
+Pros:
+
+* It is very popular. Many c++ developers already familiar with it.
+* Well maintained.
+* Do not have dependencies.
+* Can write to many sinks (console, socket, syslog)
+* DLT sink can be easily added
+* C++11
+
+Cons:
+
+* A full set of DLT features will NOT be available to app developer.
+* All primitive types (int32, float, double) will be converted to string logs.
+* LogLevel setup from DLTAppender side is not easy.  LogLevel is set on Logger level, not on Appender level.
+* DLT AppID and ContexID registration will require additional functions that breaks standard API
+
+`CAmDltWrapper (GenIVI AudioManager)`_
+""""""""""""""""""""""""""""""""""""""
+GenIVI AudioManager is a C++ program and it has it's own wrapper around DLT logging system.
+
+Pros:
+
+* Already part of GenIVI. GenIVI developers has experience with it.
+* Provide precise control over DLT message construction (supports float,int,short etc.. in DLT format)
+
+Cons:
+
+* It was designed for AudioManager, not really used outside it.
+* No documentation
+* No streaming operator support
+
+Glossary 
+--------
+
++--------+------------------------------+
+| Term   |Definition                    |
++========+==============================+
+| DLT    | Diagnostics Log and Tracing  |
++--------+------------------------------+
+| HU	 | Head Unit                    |											   
++--------+------------------------------+
+| ECU    | Electronic Control unit      |
++--------+------------------------------+
+| TCP    | Transmission Control Protocol|
```

Comment:
I think there should be a space after "Transmission Control Protocol" otherwise it is interepreted as an align right. I'm not entirely sure about it but it's also good to be consistent with spacing.